pkgs/development/tools/coder/default.nix

```
@@ -1,37 +1,93 @@
-{ buildGoModule
+{ lib
 , fetchFromGitHub
 , installShellFiles
-, lib
+, makeWrapper
+, buildGoModule
+, fetchYarnDeps
+, fixup_yarn_lock
+, pkg-config
+, nodejs
+, yarn
+, nodePackages
+, python3
+, terraform
 }:
-buildGoModule rec {
+
+let
   pname = "coder";
-  version = "0.12.5";
+  version = "0.12.4";
 
   src = fetchFromGitHub {
     owner = pname;
     repo = pname;
     rev = "v${version}";
-    hash = "sha256-tPpWj2MV2LLIOGq+RTpHpLozgqv7gBgYD3jjehRXOvk=";
+    hash = "sha256-RqdnX0oYUmJAzF3FAKHOUNMY5m8FN63c4Z/VBfJrupI=";
   };
 
-  # integration tests require network access
-  doCheck = false;
+  offlineCache = fetchYarnDeps {
+    yarnLock = src + "/site/yarn.lock";
+    sha256 = "sha256-0tZZK25bczcs4wISF5uqvbBsZ1zdKdj5GpnGSiWdNXI=";
+  };
+in
+buildGoModule rec {
+  inherit pname version src;
+
+  subPackages = [ "cmd/coder" ];
 
   vendorHash = "sha256-3SStGCDpo+AS4PM9mbXM0EjsJ/3CVFQyb/NRK9RSZ3A=";
 
-  nativeBuildInputs = [ installShellFiles ];
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/coder/coder/buildinfo.tag=${version}"
+  ];
+
+  preBuild = ''
+    export HOME=$(mktemp -d)
```

is it preferred to have?
```suggestion
    export HOME=$TEMPDIR
```